sentinel-ai-configured-agents/src/main/java/com/phonepe/sentinelai/configuredagents/ConfiguredAgent.java

```diff
@@ -56,6 +58,12 @@ public ConfiguredAgent(
         this.rootAgent = new RootAgent(agentConfiguration, agentSetup, rootAgentExtensions, availableTools);
     }
 
+    public ConfiguredAgent registerAgentMessagesPreProcessors(List<AgentMessagesPreProcessor> preProcessors) {
+        Optional.ofNullable(preProcessors)
```

Comment:
Why create another optional object here?
directly do: `rootAgent.registerAgentMessagesPreProcessors(Objects.requireNotNullElseGet(preProcessors, List::of)`